From c3573f0907dadb0a6e9933aae2a46a489abcbd48 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Thu, 16 Jun 2022 20:03:05 +0200
Subject: [PATCH 4/4] ARM: dts: qcom: add fab scaling node for ipq806x

Add fabric scaling node for ipq806x to correctly scale apps and ddr
fabric clk.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 arch/arm/boot/dts/qcom-ipq8064.dtsi | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

--- a/arch/arm/boot/dts/qcom-ipq8064.dtsi
+++ b/arch/arm/boot/dts/qcom-ipq8064.dtsi
@@ -170,6 +170,18 @@
 		};
 	};
 
+	opp_table_fab: opp_table_fab {
+		compatible = "operating-points-v2";
+
+		opp-533000000 {
+			opp-hz = /bits/ 64 <533000000>;
+		};
+
+		opp-400000000 {
+			opp-hz = /bits/ 64 <400000000>;
+		};
+	};
+
 	thermal-zones {
 		tsens_tz_sensor0 {
 			polling-delay-passive = <0>;
@@ -470,6 +482,13 @@
 			operating-points-v2 = <&opp_table_l2>;
 		};
 
+		fab-scaling {
+			compatible = "qcom,fab-scaling";
+			clocks = <&rpmcc RPM_APPS_FABRIC_A_CLK>, <&rpmcc RPM_EBI1_A_CLK>;
+			clock-names = "apps-fab-clk", "ddr-fab-clk";
+			operating-points-v2 = <&opp_table_fab>;
+		};
+
 		lpass@28100000 {
 			compatible = "qcom,lpass-cpu";
 			status = "disabled";
